screenpy/pacing.py

```diff
@@ -3,17 +3,55 @@
 (cases), and provide the gravitas (severity) of those groupings. These will
 run through all of the Narrator's adapters.
 """
-
+import functools
+import inspect
 import re
 from functools import wraps
 from typing import Any, Callable, Optional
 
 from screenpy.narration import Narrator, StdOutAdapter
+from screenpy.protocols import Answerable, Resolvable
 
 Function = Callable[..., Any]
 the_narrator: Narrator = Narrator(adapters=[StdOutAdapter()])
 
 
+# credit to https://stackoverflow.com/a/25959545/2532408
+def get_class_that_defined_method(meth: Any) -> Optional[type]:
+    """attempt to identify what class a method/function came from"""
```

Comment:
I'd like a bit more of a descriptive docstring hereand below on `function_attached_to_protocol`, but... as mentioned in other PRs, i'll be going through the docstrings shortly.

Reply:
Like more of an explanation of how it works?

Reply:
... Honestly i can't remember now, sorry for the lack of information here!

I think it had started out as a comment about the missing `Args` section in the docstring, but then i realized we were missing them everywhere, so i edited this to just ask for a bit more description.